pyvista/core/filters.py

```diff
@@ -1646,19 +1690,7 @@ def plot_over_line(dataset, pointa, pointb, resolution=None, scalars=None,
         except ImportError:
             raise ImportError('matplotlib must be available to use this filter.')
 
-        if resolution is None:
-            resolution = int(dataset.n_cells)
-        if not isinstance(resolution, int) or resolution < 0:
-            raise RuntimeError('`resolution` must be a positive integer, not {}'.format(type(resolution)))
-        # Make a line and probe the dataset
-        line = pyvista.Line(pointa, pointb, resolution=resolution)
-        sampled = line.sample(dataset)
-
-        # Get variable of interest
-        if scalars is None:
-            field, scalars = dataset.active_scalars_info
-        values = sampled.get_array(scalars)
-        distance = sampled['Distance']
+        distance, values = DataSetFilters.sample_over_line(dataset, pointa, pointb, resolution, scalars)
```

Comment:
Then this would fetch the `PolyData` line and grab the `Distance` array and use the user-specified `scalars` argument to grab the scalars that are plotted.